src/reconstruction/ppmc_cuda.cu

```diff
@@ -1,1277 +1,681 @@
+  sound_speed = hydro_utilities::Calc_Sound_Speed(cell_i.pressure, cell_i.density, gamma);
 
-      chi_1 = A * (del_d_m_i - d_6) + B * d_6;
-      chi_2 = A * (del_vx_m_i - vx_6) + B * vx_6;
-      chi_3 = A * (del_vy_m_i - vy_6) + B * vy_6;
-      chi_4 = A * (del_vz_m_i - vz_6) + B * vz_6;
-      chi_5 = A * (del_p_m_i - p_6) + B * p_6;
+  // Step 2 - Compute the left, right, centered, and van Leer differences of the primitive variables. Note that here L
+  // and R refer to locations relative to the cell center Stone Eqn 36
 
-      sum_1 += -0.5 * (d_i * chi_2 / a - chi_5 / (a * a));
-      sum_2 += 0.5 * (chi_2 - chi_5 / (a * d_i));
-      sum_5 += -0.5 * (d_i * chi_2 * a - chi_5);
-    }
-    if (lambda_0 >= 0) {
-      A = (0.5 * dtodx) * (lambda_p - lambda_0);
-      B = (1.0 / 3.0) * (dtodx) * (dtodx) * (lambda_p * lambda_p - lambda_0 * lambda_0);
-
-      chi_1 = A * (del_d_m_i - d_6) + B * d_6;
-      chi_2 = A * (del_vx_m_i - vx_6) + B * vx_6;
-      chi_3 = A * (del_vy_m_i - vy_6) + B * vy_6;
-      chi_4 = A * (del_vz_m_i - vz_6) + B * vz_6;
-      chi_5 = A * (del_p_m_i - p_6) + B * p_6;
-      #ifdef DE
-      chi_ge = A * (del_ge_m_i - ge_6) + B * ge_6;
-      #endif  // DE
-      #ifdef SCALAR
-      for (int i = 0; i < NSCALARS; i++) {
-        chi_scalar[i] = A * (del_scalar_m_i[i] - scalar_6[i]) + B * scalar_6[i];
-      }
-      #endif  // SCALAR
-
-      sum_1 += chi_1 - chi_5 / (a * a);
-      sum_3 += chi_3;
-      sum_4 += chi_4;
-      #ifdef DE
-      sum_ge += chi_ge;
-      #endif  // DE
-      #ifdef SCALAR
-      for (int i = 0; i < NSCALARS; i++) {
-        sum_scalar[i] += chi_scalar[i];
-      }
-      #endif  // SCALAR
-    }
-    if (lambda_p >= 0) {
-      A = (0.5 * dtodx) * (lambda_p - lambda_p);
-      B = (1.0 / 3.0) * (dtodx) * (dtodx) * (lambda_p * lambda_p - lambda_p * lambda_p);
-
-      chi_1 = A * (del_d_m_i - d_6) + B * d_6;
-      chi_2 = A * (del_vx_m_i - vx_6) + B * vx_6;
-      chi_3 = A * (del_vy_m_i - vy_6) + B * vy_6;
-      chi_4 = A * (del_vz_m_i - vz_6) + B * vz_6;
-      chi_5 = A * (del_p_m_i - p_6) + B * p_6;
-
-      sum_1 += 0.5 * (d_i * chi_2 / a + chi_5 / (a * a));
-      sum_2 += 0.5 * (chi_2 + chi_5 / (a * d_i));
-      sum_5 += 0.5 * (d_i * chi_2 * a + chi_5);
-    }
+  // left
+  del_L = reconstruction::Compute_Slope(cell_i, cell_im1);
 
-    // add the corrections to the initial guesses for the interface values
-    d_R += sum_1;
-    vx_R += sum_2;
-    vy_R += sum_3;
-    vz_R += sum_4;
-    p_R += sum_5;
-      #ifdef DE
-    ge_R += sum_ge;
-      #endif  // DE
-      #ifdef SCALAR
-    for (int i = 0; i < NSCALARS; i++) {
-      scalar_R[i] += sum_scalar[i];
-    }
-      #endif  // SCALAR
-
-    // right-hand interface value, i-1/2
-    sum_1 = 0;
-    sum_2 = 0;
-    sum_3 = 0;
-    sum_4 = 0;
-    sum_5 = 0;
-      #ifdef DE
-    sum_ge = 0;
-      #endif  // DE
-      #ifdef SCALAR
+  // right
+  del_R = reconstruction::Compute_Slope(cell_ip1, cell_i);
```

Comment:
I find it a little funky that these slopes mostly send the higher index first, but then the VL slope sends the lower index first. Just seems like an easy way to make a bug. I'd have written them to all take the lower index first.

Reply:
Good catch, fixed in ef1598b